app/src/main/res/layout/activity_fill_listing.xml

```diff
@@ -17,12 +17,54 @@
             android:layout_height="wrap_content"
             android:orientation="vertical">
 
-            <ImageView
-                android:id="@+id/picturePreview"
+            <TableRow
                 android:layout_width="match_parent"
-                android:layout_height="252dp"
-                android:layout_weight="1"
-                app:srcCompat="@android:drawable/screen_background_dark" />
+                android:layout_height="match_parent">
+
+                <Button
+                    android:id="@+id/setFirst"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:text="@string/move_first"
+                    android:visibility="invisible"
+                    android:textSize="12sp"
+                    android:onClick="setFirst"/>
+
+                <Button
+                    android:id="@+id/rotateLeft"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:text="@string/rotate_left"
+                    android:textSize="12sp"
+                    android:visibility="invisible"
+                    android:onClick="rotateLeft"/>
+
+                <Button
+                    android:id="@+id/deleteImage"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:text="@string/delete_image"
+                    android:textSize="12sp"
+                    android:visibility="invisible"
+                    android:onClick="deleteImage"/>
+
+                <Button
+                    android:id="@+id/modifyImage"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:onClick="modifyCurrentImage"
+                    android:text="@string/modify_image"
+                    android:visibility="invisible" />
+
+            </TableRow>
+
+            <androidx.viewpager2.widget.ViewPager2
+                android:id="@+id/viewPager2"
+                android:layout_width="match_parent"
+                android:layout_height="251dp"
+                android:layout_marginTop="16dp"
```

Comment:
Use constants, for margins especially. That way we can have uniform margins throughout the app. Constants are defined in the dimens.xml file

Reply:
Done 👍